selfservice/hook/verification.go

```diff
@@ -24,9 +24,11 @@ var _ settings.PostHookPostPersistExecutor = new(Verifier)
 type (
 	verifierDependencies interface {
 		config.Provider
-		x.CSRFTokenGeneratorProvider
+		x.CSRFProvider
```

Comment:
Why was this changed?

Reply:
Because `CSRFTokenGeneratorProvider` only contains the `GenerateCSRFToken` method, but we want to create a new CSRF token here.